### PR TITLE
Update Change basemap sample

### PR DESCRIPTION
**Pull request by vquach2404**
This PR updates the Change basemap sample to accommodate for the deprecated basemaps. Some basemaps do not have new basemap styles so I included a few of the new styles to supplement for the old ones. Please let me know what you think!

### CODE REVIEW

**Comment by vquach2404** [comment]
Ready for a review! @yo1995

File: arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift
Comment (yo1995) [inline arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift:40]: It seems this style renders without terrain.

**Review by yo1995 — APPROVED** [review]
LGTM 👍  Feel free to add others to review.
p.s. I've changed the base branch to `basemap-updates` so that we can stash the basemap changes and merge later to `v.next`.

File: arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift
Comment (vquach2404) [reply arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift:40]: I was also unsure about this style, `.osmStreetsRelief`, and `.arcGISTopographic`. They are most likely server issues and I assume that would be fixed by the release so I kept them there.

Comment (philium) [reply arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift:40]: `arcGISTerrainDetail` is documented as "A vector basemap with minimal map content like populated place names, admin and water labels with boundary lines as the reference layer. Designed to be overlaid on `AGSBasemapStyleArcGISTerrainBase` and hillshade." Therefore I don't think it is a server issue.

Comment (vquach2404) [reply arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift:40]: @philium I see. I removed it from the sample. I also corrected the names of the basemaps. Please give this a review, thanks!

Comment (philium) [reply arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift]: Thanks! Since there's no longer vector/non-vector counterparts, perhaps we could just get rid of that distinction in the basemap names? That information is available in the documentation if desired.

Comment (vquach2404) [reply arcgis-ios-sdk-samples/Maps/Change basemap/SwitchBasemapViewController.swift]: Good point! I got rid of them. Ready for your review!

**Review by philium — APPROVED** [review]

**Review by saratchandrakarumuri — COMMENTED** [review]
This sample is now prompting for credentials, which is expected as per Nimesh. But qt sampleviewer did some additional changes so that this sample won't prompt for login. Should the design be the same for iOS viewer too?

**Comment by vquach2404** [comment]
> This sample is now prompting for credentials, which is expected as per Nimesh. But qt sampleviewer did some additional changes so that this sample won't prompt for login. Should the design be the same for iOS viewer too?

Hm are you referring to the addition of the API keys? After iOS finalizes support for the API keys, this sample will only request AGOL access if API keys have not been provided. This will take into affect after #1013 is merged.

**Comment by saratchandrakarumuri** [comment]
@vquach2404 So should we update read me to mention about requirement of adding api key or authenticating? Because even with ting's branch the default behavior is not including api keys

**Comment by yo1995** [comment]
> @vquach2404 So should we update read me to mention about requirement of adding api key or authenticating? Because even with ting's branch the default behavior is not including api keys

image

Oops there seems to be some miscommunication. 
Currently we are staging all the changes to basemap onto `basemap-updates` branch.
Originally I thought #1013 will get into `v.next` soon, and after it's done I'll merge `v.next` back into `basemap-updates` branch. It turns out that the timing has changed, so I'll update #1013 tomorrow to also include API keys. 
When we finally merge `basemap-updates` into `v.next`, we can do the test then.

**Comment by vquach2404** [comment]
> When we finally merge `basemap-updates` into `v.next`, we can do the test then.

Are you saying we should hold off on testing this and instead just test the `basemap-updates` branch before merging into. `v.next`?

**Comment by yo1995** [comment]
> Are you saying we should hold off on testing this and instead just test the `basemap-updates` branch before merging into. `v.next`?

IMO For now we can do some ad-hoc testing to verify that the sample work as normal; we may wait until merging to run the automated tests like image comparison and others.

**Review by saratchandrakarumuri — APPROVED** [review]
@vquach2404 In that case we can go ahead with this PR and worry about api keys when this going into `v.next`

**Pull request merged**
